app/emoji.go

```diff
@@ -228,6 +228,18 @@ func (a *App) GetMultipleEmojiByName(names []string) ([]*model.Emoji, *model.App
 		return nil, model.NewAppError("GetMultipleEmojiByName", "api.emoji.disabled.app_error", nil, "", http.StatusNotImplemented)
 	}
 
+	if len(names) == 0 {
+		return []*model.Emoji{}, nil
+	}
+
+	if len(names) == 1 {
+		emoji, err := a.GetEmojiByName(names[0])
```

Comment:
This call use the cache, the GetMultipleByName, doesn't. Probably we can be smarter in the GetMultipleByName method and take from the cache whatever is possible and only ask for whatever is not in the cache.

Reply:
> Probably we can be smarter in the GetMultipleByName method and take from the cache whatever is possible and only ask for whatever is not in the cache.

Yes! I was going to suggest exactly that. Let's take advantage of our transparent caching layer and implement the `GetMultipleByName` in emoji_layer.go as well, and do something like `func (s *LocalCacheUserStore) GetProfileByIds`. We would first go through each emoji name in the slice and check in the cache, and only for the items not in the cache, query the store for `GetMultipleByName`, and then put back the new items in the cache.

This would also in turn benefit the `GetEmojiByName` method because new items get added in the cache.

And zero code change in the app layer :)

Reply:
I'll do the changes.

Reply:
@agnivade PTAL